develop/parachains/testing/mock-runtime.md

```diff
@@ -0,0 +1,87 @@
+---
+title: Mock Runtime for Pallet Testing
+description: Learn how to create a mock environment to test complete pallet functionalities in the Polkadot SDK, ensuring integration between pallets and system components.
+---
+
+# Mock Runtime
+
+## Introduction
+
+Testing is essential in Polkadot SDK development to ensure your blockchain operates as intended and effectively handles various potential scenarios.
+
+This guide will walk you through setting up an environment to test pallets within the runtime. Doing so lets you evaluate how different pallets, their configurations, and system components interact, ensuring your blockchain functions reliably under real-world conditions.
+
+## Configuring a Mock Runtime
+
+### Testing Module
+
+The mock runtime includes all the necessary pallets and configurations needed for testing. To simplify the process, you can create a module that integrates all components, making it easier to assess how pallets and system elements interact.
```

Comment:
Have we formally defined runtime in the glossary? it is a great candidate to go there, and you link to it.

Reply:
You can get the definition from here: 

code-wise: https://paritytech.github.io/polkadot-sdk/master/polkadot_sdk_docs/polkadot_sdk/frame_runtime/index.html 

architecturally: https://paritytech.github.io/polkadot-sdk/master/polkadot_sdk_docs/reference_docs/wasm_meta_protocol/index.html

Reply:
We do have a reference for "Runtime" in the glossary. I'll complement the actual definition with those concepts now, thanks!